src/Strategy/Python/Poetry/PoetryLock.hs

```diff
@@ -0,0 +1,168 @@
+  | MultipleObjectVersionSpec [ObjectVersion]
+  deriving (Eq, Ord, Show)
+
+newtype ObjectVersion = ObjectVersion
+  { unObjectVersion :: Text
+  }
+  deriving (Eq, Ord, Show)
+
+objectVersionCodec :: TomlCodec ObjectVersion
+objectVersionCodec =
+  ObjectVersion
+    <$> Toml.text "version" .= unObjectVersion
+
+matchTextVersion :: PoetryLockDependencySpec -> Maybe Text
+matchTextVersion (TextVersion version) = Just version
+matchTextVersion _ = Nothing
+
+matchObjectVersionSpec :: PoetryLockDependencySpec -> Maybe ObjectVersion
+matchObjectVersionSpec (ObjectVersionSpec version) = Just version
+matchObjectVersionSpec _ = Nothing
+
+matchMultipleObjectVersionSpec :: PoetryLockDependencySpec -> Maybe [ObjectVersion]
+matchMultipleObjectVersionSpec (MultipleObjectVersionSpec version) = Just version
+matchMultipleObjectVersionSpec _ = Nothing
+
+poetryLockPackagePoetryLockDependencySpecCodec :: Toml.Key -> TomlCodec PoetryLockDependencySpec
+poetryLockPackagePoetryLockDependencySpecCodec key =
+  Toml.dimatch matchTextVersion TextVersion (Toml.text key)
+    <|> Toml.dimatch matchObjectVersionSpec ObjectVersionSpec (Toml.table objectVersionCodec key)
+    <|> Toml.dimatch matchMultipleObjectVersionSpec MultipleObjectVersionSpec (Toml.list objectVersionCodec key)
+
+-- | Maps poetry lock package to map of package name and associated dependency.
+toMap :: [PoetryLockPackage] -> Map PackageName Dependency
+toMap pkgs =
+  M.fromList
+    [ ( PackageName (toLower $ unPackageName $ poetryLockPackageName pkg)
+      , Dependency
+          { dependencyType = toDepType (poetryLockPackageSource pkg)
+          , dependencyName = toDepName pkg
+          , dependencyVersion = toDepVersion pkg
+          , dependencyLocations = []
+          , dependencyEnvironments = toDepEnvironment pkg
+          , dependencyTags = M.empty
+          }
+      )
+    | pkg <- pkgs
+    ]
+  where
+    toDepName :: PoetryLockPackage -> Text
+    toDepName plp = case (poetryLockPackageSource plp) of
+      Nothing -> unPackageName $ poetryLockPackageName plp
+      Just plps -> poetryLockPackageSourceUrl plps
+
+    toDepType :: Maybe PoetryLockPackageSource -> DepType
+    toDepType Nothing = PipType
+    toDepType (Just plps) = case poetryLockPackageSourceType plps of
+      "git" -> GitType
+      "url" -> URLType
+      _ -> UserType
+
+    -- Use resolved reference (for git sources), otherwise use resolved reference
+    toDepVersion :: PoetryLockPackage -> Maybe VerConstraint
+    toDepVersion pkg = case poetryLockPackageSource pkg of
+      Nothing -> Just $ CEq $ poetryLockPackageVersion pkg
+      Just plps -> case poetryLockPackageSourceReference plps of
+        Nothing -> Just $ CEq $ poetryLockPackageVersion pkg
+        Just txt -> Just $ CEq txt
+
+    toDepEnvironment :: PoetryLockPackage -> [DepEnvironment]
+    toDepEnvironment pkg = case poetryLockPackageCategory pkg of
+      "dev" -> [EnvDevelopment]
+      _ -> [EnvProduction]
```

Comment:
We can actually cover other environments, if they pop up:
```suggestion
    toDepEnvironment :: PoetryLockPackage -> [DepEnvironment]
    toDepEnvironment pkg = case poetryLockPackageCategory pkg of
      "dev" -> [EnvDevelopment]
      "main" -> [EnvProduction]
      other -> [EnvOther other]
```

Reply:
- [ ] Include other environments